### DISCUSSION TITLE
Support storeToRefs like pinia?

### DISCUSSION THREAD

**Discussion by xoewrl** [pmndrs/zustand#2542] [Ideas] [answered]
example 
```
import {userStore} from './store.js'
const {data1, data2, method1, mehotd2} = useStore((s) => s)
```

**Comment by dai-shi** [comment]
Guessing that you want something that is not reactive. It's `useStore().getState()`.

**Reply by xoewrl** [reply]
I want to wite less code lines to save time,

Currently, I have to write n lines again and again:
```
import {userStore} from './store.js'
const data1 = useStore((s) => s.data1)
const data2 = useStore((s) => s.data2)
const data3 = useStore((s) => s.data3)
const datan = useStore((s) => s.datan)
```

Is it possibe to unpack these data{i}  using one line?
like:
```
const {data1, data2, data3, datan} = useStore((s) => s)
```

**Reply by dai-shi** [reply, marked as answer]
In that case, there are several solutions and one of them is to use createWithEqualityFn and pass shallow.

**Reply by xoewrl** [reply]
Could you provide an example? I can only find useShallow in the doc  https://docs.pmnd.rs/zustand/guides/prevent-rerenders-with-use-shallow

**Reply by xoewrl** [reply]
ok, it is in readme.md

**Reply by dai-shi** [reply]
See also: https://github.com/pmndrs/zustand/discussions/2547#discussioncomment-9400267

**Reply by xoewrl** [reply]
ok
